packages/js/api/src/framework/transformations/model-transformer-transformation.ts

```diff
@@ -74,12 +74,27 @@ export class ModelTransformerTransformation< T extends Model > implements ModelT
 	 * @return {*} The transformed properties.
 	 */
 	public toModel( properties: any ): any {
-		const val = properties[ this.property ];
+		let propertyName = this.property;
```

Comment:
I think we should investigate whether this code change is needed as a separate issue from this PR.

Reply:
I took a deep look into what is happening here and this change is necessary to ensure that a snake_case key that has a value which is an object/array is transformed correctly. 

For example:  The `line_items` key would be transformed to `lineItems` first with its value being set to the raw value of `line_items`, and this change along with adding `lineItems: 'line_items'` to  [KeyChangeTransformation<Order>](https://github.com/woocommerce/woocommerce/blob/8e98a7c1e36d5c10928a3a61c9b9237a4c2edd45/packages/js/api/src/repositories/rest/orders/transformer.ts#L56) allows the recently transformed `lineItems` key to be picked up and thats how its value gets transformed into an array of `OrderLineItems` objects

Reply:
@rrennick turns out it wasn't necessary, we just had to fix the Order model transformations that were using snake_case. I have also added a developer-friendly error if that happens again: https://github.com/woocommerce/woocommerce/pull/31665/commits/c1cf58437f29fbff9365dde7131a8eb7df89e94a